### ISSUE TITLE
Question about MonoScore calculation

### ISSUE THREAD

**Issue by anhtienng** [ExplainableML/sae-for-vlm#4]
Hi, thank you for your amazing work.

I have a question regarding the formulation of Monoscore.

In the paper, it is average by number of pairs (excluding same-image pairs).

<img width="276" height="72" alt="Image" src="https://github.com/user-attachments/assets/7a63c7ae-7433-47a4-a470-ae32279ce511" />

However, your code divide the weighted similarities by `weight_sum` that is a relevance score.
`monosemanticity = torch.where(weight_sum != 0, weighted_cosine_similarity_sum / weight_sum, torch.nan)`
(Line 73 in metric.py).

Please verify that whether the implementation is corresponding to the formulation or not. 

Thank you.

**Comment by mateuszpach** [comment]
Hi, thank you for your interest in our work!
As you noted, the implementation computes an average over the similarities weighted by the relevance scores. This is intentional. We have updated the corresponding formula in the paper to reflect this (we now divide by the sum of the weights instead of the number of samples).
Thank you for bringing this to our attention!

**Comment by anhtienng** [comment]
Thank you!
